### ISSUE TITLE
Type synonyms not available when using data-dependencies

### ISSUE THREAD

**Issue by jberthold-da** [digital-asset/daml#6306]
We want to use data dependencies so we can use dependencies between a (very stable) package compiled with one SDK version and a second package of code expected to change more often (including SDK upgrades).
Unfortunately, type synonyms (`type X = ...`) from the stable package cannot be imported in the second package. The type that the synonym stands for can be used, but not the synonym, as in this example:

Package1 has the following DAML code
```
module Package1File where
type T = D Text
data D a = D with a : a
def1 : T
def1 = D "def1"
```
Package2 imports that module and uses its contents:
```
module Package2File where
import Package1File -- from package 1
def2 : D Text -- OK
-- def2 : T -- compilation error
def2 = def1
```
and Package2 `daml.yaml` specifies a data dependency
```
sdk-version: ...
name: package2
version: 0.0.0
parties:
  - 00001
dependencies:
- daml-prim
- daml-stdlib
data-dependencies:
- ../package1/.daml/dist/package1-0.0.0.dar
source: src/
```
Is it possible to make the type synonym available?

**Comment by cocreature** [comment]
Fixed in #6894, thanks @sofiafaro-da!

**Comment by sofiafaro-da** [comment]
Should be fixed in #6894. That PR makes normal type synonyms available in the DALF, and can pick them up via data-dependencies. (Pre-existing DALFs won't expose type synonyms, they need to be recompiled.)
